diff --git a/SConstruct b/SConstruct
index 9a1c888..0091fb5 100644
--- a/SConstruct
+++ b/SConstruct
@@ -343,13 +343,6 @@ if GetOption('help'):
 
 variantDir = get_variant_dir()
 
-def printLocalInfo():
-    import sys, SCons
-    print( "scons version: " + SCons.__version__ )
-    print( "python version: " + " ".join( [ `i` for i in sys.version_info ] ) )
-
-printLocalInfo()
-
 boostLibs = [ "thread" , "filesystem" , "program_options", "system" ]
 
 onlyServer = len( COMMAND_LINE_TARGETS ) == 0 or ( len( COMMAND_LINE_TARGETS ) == 1 and str( COMMAND_LINE_TARGETS[0] ) in [ "mongod" , "mongos" , "test" ] )
diff --git a/site_scons/libdeps.py b/site_scons/libdeps.py
index e17a4fa..3cbaaa5 100644
--- a/site_scons/libdeps.py
+++ b/site_scons/libdeps.py
@@ -56,6 +56,8 @@ import SCons.Errors
 import SCons.Scanner
 import SCons.Util
 
+import json
+
 libdeps_env_var = 'LIBDEPS'
 syslibdeps_env_var = 'SYSLIBDEPS'
 missing_syslibdep = 'MISSING_LIBDEP_'
@@ -100,6 +102,7 @@ def __compute_libdeps(node):
 
     env = node.get_env()
     deps = set()
+    topleveldeps = set()
     node.attributes.libdeps_exploring = True
     try:
         try:
@@ -113,9 +116,22 @@ def __compute_libdeps(node):
             if len(e.cycle_nodes) == 1 or e.cycle_nodes[0] != e.cycle_nodes[-1]:
                 e.cycle_nodes.append(node)
                 raise
+
+        try:
+            for child in env.Flatten(getattr(node.attributes, 'libdeps_direct', [])):
+                if not child:
+                    continue
+                topleveldeps.add(child)
+
+        except DependencyCycleError, e:
+            if len(e.cycle_nodes) == 1 or e.cycle_nodes[0] != e.cycle_nodes[-1]:
+                e.cycle_nodes.append(node)
+                raise
     finally:
         node.attributes.libdeps_exploring = False
 
+    print({ '_id': str(node), 'type': 'special', 'libdeps': [ str(dep) for dep in topleveldeps ] })
+
     return deps
 
 def __get_syslibdeps(node):
@@ -256,7 +272,7 @@ def setup_environment(env):
         env['_LIBDEPS'] = '$_LIBDEPS_LIBS'
 
     # TODO: remove this
-    # this is a horrible horrible hack for 
+    # this is a horrible horrible hack for
     # for 32-bit solaris
     if "uname" in dir(os) and os.uname()[1] == "sun32b":
         env['_LIBDEPS_LIBS'] = get_libdeps_special_sun
